anvill/python/anvill/binja/bnprogram.py

```diff
@@ -121,6 +187,24 @@ def get_function_impl(self, address):
             arg_type = self.type_cache.get(var_type)
 
             if source_type == bn.VariableSourceType.RegisterVariableSourceType:
+
+                # For some functions binary ninja identifies the function type incorrectly. The
+                # register allocation in such cases violate calling convention.
+                #
+                # https://github.com/Vector35/binaryninja-api/issues/2399
+                # https://github.com/Vector35/binaryninja-api/issues/2400
+                #
+                # Identify the storage register and discard the parameter variable
+                # if they does not follow calling convention
+                # e.g: int32_t main(int32_t arg1, void* arg2, int128_t arg3 @ q1, int64_t arg4 @ q2)
+                #
+                storage_reg_name = self._bv.arch.get_reg_name(var.storage)
+                if not (
+                    storage_reg_name in calling_conv.int_arg_reg
+                    or storage_reg_name in calling_conv.float_arg_reg
+                ):
+                    continue
+
                 if (
```

Comment:
We need to change a lot of / remove a lot of the below, I assume? That is, it seems like we should only use `storage_reg_name`, and if not, we need to be consistent and make sure that if there was a `storage_reg_name` for some prior parameter, that we don't permit that register name to be used anymore. I.e. the `CallingConvention` should have some kind of used registers tracking, to prevent accidental reuse.

Reply:
We should throw an exception for that "invalid void" instead of admitting a bad register.

Reply:
Also, if you find such an example, then report an issue to v35.

Reply:
ref:
https://github.com/Vector35/binaryninja-api/issues/2399
https://github.com/Vector35/binaryninja-api/issues/2400